### DISCUSSION TITLE
Shooting top layer on green screen then overlay on background?

### DISCUSSION THREAD

**Discussion by TheNettletons17** [ImageMagick/ImageMagick#4384] [Digital Image Processing]
I am trying to come up with the process on a Windows Commandline(batch) for taking a top layer shot against a greenscreen, remove all reasonable green values and then overlap that on a background file.
Top Layer:
![tsp_20000002](https://user-images.githubusercontent.com/92743042/137765459-2069ce0e-3378-4202-abbf-fc231f70a08b.jpg)
Background:
![trees](https://user-images.githubusercontent.com/92743042/137765489-5257740a-e4ef-49b7-8539-c37318a9421f.jpg)

Here's the code:
magick convert tests\20000002.jpg -format png -transparent "#11FF3A" tests\transparent50b.png
which produces not quite what I want:
![transparent50b](https://user-images.githubusercontent.com/92743042/137765822-8d8b8123-d705-4a8b-b7f4-c3fa6ddea3d5.png)

The goal is something like this:
![image](https://user-images.githubusercontent.com/92743042/137766335-154f91de-6bbf-4d00-b127-403876d4e427.png)

1.) Is there a switch like -FUZZ for getting kinda close to the green values so I get them all? (I realize that might get green that the subject is wearing)
2.) Do I need to do this in multiple passes or all in one?

Thanks!!

**Comment by fmw42** [comment]
What version of Imagemagick?

**Reply by fmw42** [reply]
You can use -fuzz with -transparent!  Have you tried that?

convert green.jpg -fuzz 35% -transparent "#11FF3A" x.png

![x](https://user-images.githubusercontent.com/13458218/137767873-44e5113b-fb23-4f1d-8cf7-64769d3e0156.png)

**Reply by fmw42** [reply]
Then you can feather the result onto the background.

convert back.jpg ( green.jpg -background none -gravity south -extent 1175x620 ) ^
( -clone 1 -fuzz 35% -transparent "#11FF3A" -alpha extract -blur 0x2 -level 50x100% ^ 
-background black -gravity south -extent 1175x620 ) ^
-compose over -composite result.jpg

![result](https://user-images.githubusercontent.com/13458218/137769668-6822a606-2604-473f-a57f-0daccc75411c.jpg)

If in .bat file, double the % to %%

**Reply by TheNettletons17** [reply]
I'm using 7.1 and I've tried the -fuzz with no affect.  
magick convert tests\20000002.jpg -format png -fuzz 90% -transparent "#11FF3A"  tests\transparent90b.png
No change:
![transparent90b](https://user-images.githubusercontent.com/92743042/137775352-7ec6f180-786a-4a48-a1bd-2978ef80b522.png)

**Reply by fmw42** [reply]
try

`magick tests\20000002.jpg -format png -fuzz 35% -transparent "#11FF3A" tests\transparent90b.png`

if in .bat, the double the % to %%

Note for IM 7, use magick, not magick convert and not convert.

**Reply by fmw42** [reply]
You would generally be better if you did not use JPG due to lossy compression. So if possible use PNG or TIFF for your images (not converted from JPG).

**Comment by aral** [comment]
Links to legacy answers that might help:

  - https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=14394
  - https://legacy.imagemagick.org/Usage/photos/#chroma_key
  - https://legacy.imagemagick.org/Usage/masking/#bg_remove

**Comment by fmw42** [comment]
You would be better thresholding appropriate combination of LAB channels, i.e. the product of the negated B with A channels. See for example my bash unix shell script for green screening at http://www.fmwconcepts.com/imagemagick/greenscreen/index.php. You can run that on Windows 10 Unix as well as Linux and Mac OSX.
